src/pages/Search.js

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
-import { useParams } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { Redirect, useParams, withRouter } from 'react-router-dom';
+import SubredditForm from '../components/SubredditForm';
 import defaultSubreddit from '../helpers/constants';
 
+
 const Search = () => {
-  const { subreddit = defaultSubreddit } = useParams();
-  return (
+  const { subredditQuery } = useParams();
+  const [subreddit, setSubreddit] = useState(subredditQuery || defaultSubreddit);
+  const [route, setRoute] = useState(null);
+
+  const handleSubmit = (evt) => {
+    setRoute(`/search/${subreddit}`);
+    evt.preventDefault();
+  };
+
+  useEffect(() => {
+    document.getElementById('subreddit').value = subredditQuery;
+  }, [subredditQuery]);
+
+  const handleChange = (evt) => setSubreddit(evt.target.value);
+
+  const searchPage = (
     <main>
-      Search placeholder:
-      { subreddit }
+      <SubredditForm
+        subreddit={subreddit}
+        handleSubmit={handleSubmit}
+        handleChange={handleChange}
+      />
     </main>
   );
-};
 
+  if (route) {
```

Comment:
This is an interesting approach and works fine. It looks a bit like a workaround though. With the `useHistory` hook I mentioned above you won't need this